Automattic/wp-calypso
# Framework: Remove unused ProductsList

#### Changes proposed in this Pull Request

After #56268 and #55665, the `ProductsList` emitter store is now unused and can be removed. 

It's also one of the last usages of the emitter mixin, which would be another ancient artifact we'd want to get rid of in the future.

#### Testing instructions

Verify the removed code is not in use. There should be no bundle size changes.

Fixes #48406.
